easybuild/framework/easyconfig/parser.py

```diff
@@ -131,6 +137,46 @@ def _read(self, filename=None):
             msg = 'rawcontent is not basestring: type %s, content %s' % (type(self.rawcontent), self.rawcontent)
             raise EasyBuildError("Unexpected result for raw content: %s", msg)
 
+    def _extract_comments(self):
+        """Extract comments from raw content."""
+        # Keep track of comments and their location (top of easyconfig, key they are intended for, line they are on
+        # discriminate between header comments (top of easyconfig file), single-line comments (at end of line) and other
+        # At the moment there is no support for inline comments on lines that don't contain the key value
+
+        self.comments = {
+            'header' : [],
+            'inline' : dict(),
+            'above' : dict(),
+         }
+
+        raw = self.rawcontent.split('\n')
+        header = True
+
+        i = 0
+        while i<len(raw):
```

Comment:
it's not a for loop because the counter needs to be incremented more if there are multiline comments (the nested loop on line 164)